ui/components/multichain/select-action-modal/select-action-modal.js

```diff
@@ -0,0 +1,239 @@
+                    chain_id: chainId,
+                    token_symbol: defaultSwapsToken,
+                  },
+                });
+                onClose();
+              }}
+            />
+            ///: END:ONLY_INCLUDE_IN
+          }
+          {
+            ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+            <SelectActionModalItem
+              actionIcon={IconName.Stake}
+              showIcon
+              primaryText={t('stake')}
+              secondaryText={t('stakeDescription')}
+              onClick={() => {
+                stakingEvent();
+                global.platform.openTab({
+                  url: MMI_STAKE_WEBSITE,
+                });
+                onClose();
+              }}
+            />
+            ///: END:ONLY_INCLUDE_IN
+          }
+          <SelectActionModalItem
+            actionIcon={IconName.SwapHorizontal}
+            primaryText={t('swap')}
+            data-testid="select-action-swap"
+            secondaryText={t('swapDescription')}
+            isDisabled={!isSwapsChain}
+            tooltipTitle={t('swapDisabled')}
+            onClick={() => {
+              ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+              global.platform.openTab({
+                url: MMI_SWAPS_URL,
+              });
+              ///: END:ONLY_INCLUDE_IN
+
+              ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
+              if (isSwapsChain) {
+                trackEvent({
+                  event: MetaMetricsEventName.NavSwapButtonClicked,
+                  category: MetaMetricsEventCategory.Swaps,
+                  properties: {
+                    token_symbol: 'ETH',
+                    location: MetaMetricsSwapsEventSource.MainView,
+                    text: 'Swap',
+                    chain_id: chainId,
+                  },
+                });
+                dispatch(setSwapsFromToken(defaultSwapsToken));
+                if (usingHardwareWallet) {
+                  global.platform.openExtensionInBrowser(BUILD_QUOTE_ROUTE);
+                } else {
+                  history.push(BUILD_QUOTE_ROUTE);
+                }
+              }
+              ///: END:ONLY_INCLUDE_IN
+              onClose();
+            }}
+          />
+          <SelectActionModalItem
+            actionIcon={IconName.Arrow2UpRight}
+            primaryText={t('send')}
+            secondaryText={t('sendDescription')}
+            data-testid="select-action-send"
+            onClick={() => {
+              trackEvent({
+                event: MetaMetricsEventName.NavSendButtonClicked,
+                category: MetaMetricsEventCategory.Navigation,
+                properties: {
+                  token_symbol: 'ETH',
+                  location: 'Home',
+                  text: 'Send',
+                  chain_id: chainId,
+                },
+              });
+              dispatch(
+                startNewDraftTransaction({ type: AssetType.native }),
+              ).then(() => {
```

Comment:
I think we should try to get into the habit of using `async/await` in our new code, avoiding `then`'s.

Reply:
Nice suggestion! Updated here https://github.com/MetaMask/metamask-extension/pull/20559/commits/8d602620ce1bdf9f6b57d9e53f40aa8f8cb9580e